sympy/physics/control/control_plots.py

```diff
@@ -0,0 +1,160 @@
+
+    num_poly = Poly(system.num, system.var).all_coeffs()
+    den_poly = Poly(system.den, system.var).all_coeffs()
+
+    num_poly = np.array(num_poly, dtype=np.float64)
+    den_poly = np.array(den_poly, dtype=np.float64)
+
+    zeros, poles = _find_roots(num_poly, den_poly)
+
+    zero_real = np.real(zeros)
+    zero_imag = np.imag(zeros)
+
+    pole_real = np.real(poles)
+    pole_imag = np.imag(poles)
+
+    plt.plot(pole_real, pole_imag, 'x', mfc='none', markersize=10)
+    plt.plot(zero_real, zero_imag, 'o', markersize=7)
+    plt.xlabel('Real')
+    plt.ylabel('Imaginary')
+    plt.title('Poles and Zeros')
+    if grid:
+        plt.grid()
+    plt.axhline(0, color='black')
+    plt.axvline(0, color='black')
+    if show:
+        plt.show()
+        return
+    return plt
+
+
+def step_response(system, show_input=False, colour='r', show=True, grid=True, upper_limit=6, **kwargs):
+    r"""
+    Return the unit step response of a continuous-time system.
+    """
+    x = Symbol("x")
+    expr = system.to_expr()/(system.var)
+    y = inverse_laplace_transform(expr, system.var, x)
+    return plot(y, (x, 0, upper_limit), show=show, title="Unit Step Response", \
+        xlabel="Time (s)", ylabel="Amplitude")
+
+
+def impulse_response(system, show_input=False, colour='r', show=True, grid=True, upper_limit=6, **kwargs):
+    r"""
+    Return the unit impulse response (Input is the Dirac-Delta Function) of a
+    continuous-time system.
+    """
+    x = Symbol("x")
+    expr = system.to_expr()
+    y = inverse_laplace_transform(expr, system.var, x)
+    return plot(y, (x, 0, 6), show=True, title="Impulse Response", \
+        xlabel="Time (s)", ylabel="Amplitude")
+
+
+def ramp_response(system, slope=1, show_input=False, colour='r', show=True, grid=True, upper_limit=6, **kwargs):
+    r"""
+    Return the unit impulse response (Input is the Dirac-Delta Function) of a
+    continuous-time system.
+    """
+    x = Symbol("x")
+    expr = (slope*system.to_expr())/((system.var)**2)
+    y = inverse_laplace_transform(expr, system.var, x)
+    return plot(y, (x, 0, upper_limit), show=show, title="Ramp Response", \
+        xlabel="Time (s)", ylabel="Amplitude")
+
+
+def bode_plot(system, initial_exp=-5, final_exp=5, show=True, **kwargs):
+    r"""
+    Plot a Bode plot of a continuous-time system.
+    """
+    expr = system.to_expr()
+    w = Symbol("w", real=True)
+    w_expr = expr.subs({system.var: I*w})
+    real, imag = w_expr.as_real_imag()
+
+    mag = 20*log(sqrt(real**2 + imag**2), 10)
+
+    mag_plot = plot(mag, (w, 10**initial_exp, 10**final_exp), xscale='log', title="Bode Plot (Magnitude)", \
+        xlabel="Frequency (Hz) [Log Scale]", ylabel="Magnitude (dB)", show=False)
+
+    phase = arg(w_expr)
+
+    phase_plot = plot(phase, (w, 10**initial_exp, 10**final_exp), xscale='log', title="Bode Plot (Phase)", \
+        xlabel="Frequency (Hz) [Log Scale]", ylabel="Phase (rad)", show=False)
```

Comment:
Unfortunately, I don't think this is so easy. Can you give me some links which you followed so that I could check the correctness?

Reply:
> Can you give me some links which you followed so that I could check the correctness?

https://en.wikipedia.org/wiki/Bode_plot
https://www.youtube.com/watch?v=_eh1conN6YM
https://www.youtube.com/watch?v=CSAp9ooQRT0

I didn't look at someone's implementation. I just read the Wikipedia page saw a couple of related videos/univ lectures. I tested it with 10s of different examples which I could find online and all the plots matched exactly. There might be few vulnerabilities, if you find any particular case just let me know.

> Unfortunately, I don't think this is so easy.

The point here is not about being easy. It is the **definition** of the Bode plot. This is how it is defined. I really don't get that how can it be wrong if the calculation of points is done by the sole definition of the plot.

Reply:
> I really don't get that how can it be wrong if the calculation of points is done by the sole definition of the plot.

I did not say that you are wrong. I just had some doubts and wanted to verify the correctness myself so I asked you to provide me some links.

> I tested it with 10s of different examples which I could find online and all the plots matched exactly. 

Thanks!

Reply:
> There might be few vulnerabilities, if you find any particular case just let me know.

@akshanshbhatt it is your job to figure these out and show us that you've fully though this through. Right now, it doesn't look like you have.